Sprint 1/step 1

EldenLordCommander opened the pull request:
Пул-реквест заготовки, для промежуточной проверки, чтобы понять в правильном ли направлении веду работу

EldenLordCommander committed:
add json-data

Добавил в App файл json, с которого будут браться пропсы

EldenLordCommander committed:
Заготовка

Добавил компоненты, стили, постарался сверстать вариант, похожий на макет

gennady-bars commented on src/components/app/App.tsx:
```diff
@@ -0,0 +1,258 @@
+import BurgerConstructor from '../burger-constructor/burger-constructor';
+
+function App() {
+   const data = [
```
моковые данные нужно вынести в `utils` в свою папку, чтобы не засорять код компонентов

gennady-bars commented on src/components/app/App.tsx:
```diff
@@ -0,0 +1,258 @@
+
+   return (
+      <div className="App">
+         <header>
```
тег `header` должен быть внутри `AppHeader`. В этом его суть

gennady-bars commented on src/components/app/App.tsx:
```diff
@@ -0,0 +1,258 @@
+         <main className={appStyles.main}>
+            <div className={appStyles.leftColumn}>
+               <span>
+                  <p className="text text_type_main-large" style={{ textAlign: 'left' }}>Соберите бургер</p>
```
Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-constructor/burger-constructor.js:
```diff
@@ -0,0 +1,81 @@
+    return (
+        <section>
+            <section>
+                <div style={{ marginTop: '16px', marginBottom: '16px' }} className={constructorStyles.constructorIngregient}>
```
Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-constructor/burger-constructor.js:
```diff
@@ -0,0 +1,81 @@
+        <section>
+            <section>
+                <div style={{ marginTop: '16px', marginBottom: '16px' }} className={constructorStyles.constructorIngregient}>
+                    <div style={{ opacity: 0 }}>
```
Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-constructor/burger-constructor.js:
```diff
@@ -0,0 +1,81 @@
+                    leftIngredients.map((item, key) =>
+                    (
+
+                        <div className={constructorStyles.constructorIngregient} style={{ marginTop: '16px', marginBottom: '16px' }} key={key}>
```
Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-constructor/burger-constructor.js:
```diff
@@ -0,0 +1,81 @@
+                }
+            </section>
+            <section>
+                <div style={{ marginTop: '16px', marginBottom: '16px' }} className={constructorStyles.constructorIngregient}>
```
Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-constructor/burger-constructor.js:
```diff
@@ -0,0 +1,81 @@
+            <section>
+                <div style={{ marginTop: '16px', marginBottom: '16px' }} className={constructorStyles.constructorIngregient}>
+                    <div style={{ opacity: 0 }}>
+                        <DragIcon type="primary" />
```
булкам не нужно указывать `DragIcon` и делать их невидимыми. В этом нет смысла

gennady-bars commented on src/components/burger-constructor/burger-constructor.js:
```diff
@@ -0,0 +1,81 @@
+            <section>
+                <div style={{ marginTop: '16px', marginBottom: '16px' }} className={constructorStyles.constructorIngregient}>
+                    <div style={{ opacity: 0 }}>
+                        <DragIcon type="primary" />
```
булкам не нужно указывать `DragIcon` и делать их невидимыми. В этом нет смысла

gennady-bars commented on src/components/burger-ingredients/burger-ingredients.js:
```diff
@@ -0,0 +1,92 @@
+    return (
+        <section className={ingredientStyle.mainIngredientList}>
+            <div className={ingredientStyle.ingredientList}>
+                <p className="text text_type_main-medium" style={{ textAlign: 'left' }}>Булки</p>
```
* Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-ingredients/burger-ingredients.js:
```diff
@@ -0,0 +1,92 @@
+                }
+            </div>
+            <div className={ingredientStyle.ingredientList}>
+                <p className="text text_type_main-medium" style={{ textAlign: 'left' }}>Соусы</p>
```
* Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-ingredients/burger-ingredients.js:
```diff
@@ -0,0 +1,92 @@
+                }
+            </div>
+            <div className={ingredientStyle.ingredientList}>
+                <p className="text text_type_main-medium" style={{ textAlign: 'left' }}>Начинка</p>
```
* Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/tab-header/tab-header.js:
```diff
@@ -0,0 +1,22 @@
+function TabHeader () {
+    const [current, setCurrent] = React.useState('one')
+    return (
+      <div style={{ display: 'flex' }}>
```
* Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/app/App.tsx:
```diff
@@ -0,0 +1,258 @@
+         </header>
+         <main className={appStyles.main}>
+            <div className={appStyles.leftColumn}>
+               <span>
```
`<span>`

пустой `<span>` не нужен вокруг `p`. Он ничего не дает


и `Соберите бургер` является главным важным текстом на странице, поэтому по семантике нужно указать в теге `h1`

gennady-bars replied:
```jsx


               <span>
                  <p className="text text_type_main-large" style={{ textAlign: 'left' }}>Соберите бургер</p>
               </span>
               <TabHeader />
               <div className={appStyles.ingredientsList}>
                  <BurgerIngredients data={data} />
               </div>

```

Вам нужно упаковать всю эту верстку в `BurgerIngredients`. Это очистит код `App`

gennady-bars commented on src/components/app/App.tsx:
```diff
@@ -0,0 +1,258 @@
+            </div>
+            <div className={appStyles.rightColumn}>
+               <BurgerConstructor data={data}  />
+               <div className={appStyles.checkout}>
```
```jsx

               <BurgerConstructor data={data}  />
               <div className={appStyles.checkout}>
                  <div className={appStyles.price}>
                     <span className="text text_type_main-large">0</span>
                     <CurrencyIcon type="primary" />
                  </div>
                  <Button type="primary" size="large" htmlType={'button'}>
                     Оформить заказ
                  </Button>
               </div>

```

Вам нужно упаковать всю эту верстку в `BurgerConstructor`. Это очистит код `App`


В итоге остается аккуратный код


```jsx

      <main className={appStyles.main}>
        <div className={appStyles.leftColumn}>
          <BurgerIngredients data={data} />
        </div>
        <div className={appStyles.rightColumn}>
          <BurgerConstructor data={data} />
        </div>
      </main>


```


а еще лучше и `div` тоже туда убрать


```jsx

      <main className={appStyles.main}>
        <BurgerIngredients data={data} />
        <BurgerConstructor data={data} />
      </main>

```

gennady-bars commented on src/components/app-header/app-header.js:
```diff
@@ -0,0 +1,33 @@
+
+import { BurgerIcon, Logo, ListIcon, ProfileIcon } from '@ya.praktikum/react-developer-burger-ui-components'
+
+class AppHeader extends React.Component {
```
**Можно лучше**

 **классовые компоненты** редко используются в современных проектах. Особенно, если в них нет стейта `state`. Лучше всегда использовать функциональные + хуки

gennady-bars commented on src/components/burger-constructor/burger-constructor.js:
```diff
@@ -0,0 +1,81 @@
+
+BurgerConstructor.propTypes ={
+    data : PropTypes.arrayOf(PropTypes.shape({
+            name : PropTypes.string.isRequired,   
```
тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно

gennady-bars commented on src/components/burger-ingredients/burger-ingredients.js:
```diff
@@ -0,0 +1,92 @@
+
+BurgerIngredients.propTypes ={
+    data : PropTypes.arrayOf(PropTypes.shape({
+        name : PropTypes.string.isRequired,   
```
тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно

gennady-bars commented on src/components/burger-ingredients/burger-ingredients.js:
```diff
@@ -0,0 +1,92 @@
+                        }
+                    })
+                    .map((item, key) => (
+                        <div className={ingredientStyle.ingredient} key={key}>
```
`key={key}`

Для атрибута `key` лучше всего использовать уникальный `id` : у карточки как раз есть такой только с нижним подчеркиванием `_id`
[Почему индекс не нужно использовать, можно прочитать тут:]( https://medium.com/russian/%D0%B8%D0%BD%D0%B4%D0%B5%D0%BA%D1%81-%D0%B2-%D0%BA%D0%B0%D1%87%D0%B5%D1%81%D1%82%D0%B2%D0%B5-%D0%BA%D0%BB%D1%8E%D1%87%D0%B0-%D0%B0%D0%BD%D1%82%D0%B8-%D0%BF%D0%B0%D1%82%D1%82%D0%B5%D1%80%D0%BD-56bab8ada309)

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная, но есть некоторые недочеты: 


* Файл `App.test.js` нужно удалить, так как он не относится к проекту
* лишний `App.css` нужно удалить
* ` App.tsx` нужно сделать с маленькой буквы, так как по заданию нужно называть файлы в `kebab-case`, 
* В верхней булке нужно добавлять текст `верх`, а в нижней - `низ`. Посмотрите в фигме, пожалуйста
* Верхняя и нижняя булки должны быть всегда **одинаковыми** по заданию
* моковые данные нужно вынести в `utils` в свою папку, чтобы не засорять код компонентов
* тег `header` должен быть внутри `AppHeader`. В этом его суть
* Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.
* булкам не нужно указывать `DragIcon` и делать их невидимыми. В этом нет смысла
*  `Соберите бургер` является главным важным текстом на странице, поэтому по семантике нужно указать в теге `h1`
* тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно
* Для атрибута `key` лучше всего использовать уникальный `id` : у карточки как раз есть такой только с нижним подчеркиванием `_id`
* блок с заказом (бургером) нужно сделать чуть больше, чтобы тест в его ингредиентах помещался в длину. Посмотрите в фигме






**Можно лучше**

* **классовые компоненты** редко используются в современных проектах. Особенно, если в них нет стейта `state`. Лучше всегда использовать функциональные + хуки


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

EldenLordCommander committed:
Исправление недочетов

Удаление лишних файлов, перенос системных данных в utils, избавление от styles, различные правки

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.

p.s. теперь нужно сделать 2й этап с попапами и прислать в этом же тикете

* проект не запускается. Скрин ошибки https://disk.yandex.ru/i/P0k1SPJZMWKKdg . Это из-за `App.tsx`, который не исправился на `app.tsx`. Я вручную исправляю и все запускается
* Видимо, **кэш гита** не отслеживает регистр файлов. Для гита без разницы, с маленькой буквы или с большой сделан файл. Он считает их одинаковыми. Попробуйте очистить кэш гита. Посоветуйтесь с наставником или посмотрите по этой ссылке, как это сделать https://coderoad.ru/41863484/%D0%9E%D1%87%D0%B8%D1%81%D1%82%D0%B8%D1%82%D1%8C-%D0%BB%D0%BE%D0%BA%D0%B0%D0%BB%D1%8C%D0%BD%D1%8B%D0%B9-%D0%BA%D1%8D%D1%88-git

EldenLordCommander committed:
rename App to app
